src/librustc_lint/bad_style.rs

```diff
@@ -342,6 +342,17 @@ impl NonUpperCaseGlobals {
     fn check_upper_case(cx: &LateContext, sort: &str, name: ast::Name, span: Span) {
         if name.as_str().chars().any(|c| c.is_lowercase()) {
             let uc = NonSnakeCase::to_snake_case(&name.as_str()).to_uppercase();
+
+            let start = cx.tcx.sess.codemap().span_to_snippet(span)
+                .map(|snippet| snippet.find(name.as_str().chars().as_str()).unwrap_or(0))
```

Comment:
In order to avoid the issue mentioned below, you should probably find the end of the first occurrence of whitespace, and only then `find` the `name`.

Reply:
If adding a span to AST, then the above comment is moot.